File: DSA-3/Hashing/check-if-two-arrays-are-equal-or-not.cpp
/********

https://www.geeksforgeeks.org/check-if-two-arrays-are-equal-or-not/

*******/

#include <bits/stdc++.h>
using namespace std;

bool compareArray(long long int arr1[],long long int arr2[],long n){
    
    
	    map<long long int,long long int> m;
	    for(long i=0; i<n; i++){
	       
	        if(m.find(arr1[i]) == m.end()){
	            m[arr1[i]] = 1;
	        }else{
	            m[arr1[i]]++;
	        }
	    }
	    
	    for(long i=0; i<n; i++){
	       
	        if(m.find(arr2[i]) == m.end()){
	           
	             return false;
	            
	        }
	        if(m[arr2[i]] == 0){
	            return false;
	        }
	        
	        m[arr2[i]]--;
	    }
	    
	    return true;
}

int main() {
	//code
	int t;
	cin>>t;
	
	while(t--){
	    long n;
	    cin>>n;
	    long long int arr1[n] , arr2[n];
	    
	    for(long i=0; i<n; i++){
	      cin>>arr1[i];  
	    }
	    
	    for(long i=0; i<n; i++){
	        cin>>arr2[i];
	    }
	    
	    if(compareArray(arr1,arr2,n)){
	        
	        cout<<1<<endl;
	    }else{
	        cout<<0<<endl;
	    }
	    
	}
	return 0;
}

File: Company Wise Practice/Goldman Sachs/check-if-two-arrays-are-equal-or-not.java
/*package whatever //do not write package name here */
/*******

https://www.geeksforgeeks.org/check-if-two-arrays-are-equal-or-not/

********/
import java.util.*;
import java.lang.*;
import java.io.*;

class GFG {
	public static void main (String[] args) {
		//code
		Scanner sc = new Scanner(System.in);
		
		Integer t = sc.nextInt();
		
		while(t-- > 0){
		    
		    Integer n = sc.nextInt();
		    
		    Map<Long,Integer> map = new HashMap<>();
		    
		    for(int i=0; i<n; i++){
		        Long temp = sc.nextLong();
		        
		        if(map.containsKey(temp)){
		            map.put(temp,map.get(temp) + 1);
		        }else{
		            map.put(temp,1);
		        }
		        
		    }
		    
		    boolean flag = true;
		    
		    long arr[] = new long[n];
		    
		    for(int i=0; i<n; i++){
		      arr[i] = sc.nextLong();  
		    }
		    
		    
		    for(int i=0; i<n; i++){
		        
		        long temp = arr[i];
		        
		        if(map.containsKey(temp)){
		            map.put(temp,map.get(temp) - 1);
		            
		            if(map.get(temp) == 0){
		                map.remove(temp);
		            }
		        }else{
		            flag = false;
		            break;
		        }
		    }
		    
		    if(map.size() == 0){
		        flag = true;
		    }
		    
		    if(flag){
		        System.out.println("1");
		    }else{
		        System.out.println("0");
		    }
		}
		
		sc.close();
	}
}